src/toolchain.rs

```diff
@@ -371,6 +376,42 @@ impl<'a> Toolchain<'a> {
         Ok(None)
     }
 
+    fn maybe_do_rust_analyzer_fallback(&self, binary: &str) -> anyhow::Result<Option<Command>> {
```

Comment:
Add a docstsring that briefly describes why this is useful/necessary and what the goal/semantics of the function should be?

Reply:
@djc Nice point, and fixed!